src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs

```diff
@@ -285,22 +285,22 @@ private int CalculateMemoryNoBoundsCheck(int oldOffset, int totalSize, out int a
             newCategoryEntryPointer->NextCategoryOffset = 0;
             newCategoryEntryPointer->FirstInstanceOffset = (int)((long)newInstanceEntryPointer - _baseAddress);
             newCategoryEntryPointer->CategoryNameOffset = (int)(nextPtr - _baseAddress);
-            SafeMarshalCopy(_categoryName, checked((IntPtr)nextPtr));
+            SafeMarshalCopy(_categoryName, (nint)nextPtr);
```

Comment:
Could you please change the argument type of SafeMarshalCopy to void* and delete these casts